packages/serverpod/lib/src/util/string_extension.dart

```diff
@@ -0,0 +1,27 @@
+/// Extension methods for the [String] class.
+extension StringExtension on String? {
+  /// Returns true if the string is null or empty.
+  bool get isNullOrEmpty => this == null || this!.trim().isEmpty;
+
+  /// Returns true if the string is not null or empty.
+  bool get isNotNullOrEmpty => !isNullOrEmpty;
+
+  /// Returns the substring of the string, or an empty string if the string is
+  /// null or empty.
+  String subString(int start, [int? end]) =>
```

Comment:
Suggestion: Remove override of native method or make the naming more specific.

For new developers wanting to contribute to the framework it is good to have as few surprises as possible (and us once we have forgotten about his override). Giving new behavior to built in language methods is a sure way to throw curveballs at people.

Reply:
Yup this was updated. please re-review this